src/linker/Linker.Steps/MarkStep.cs

```diff
@@ -333,11 +350,54 @@ void Process ()
 						_context.MarkingHelpers.MarkExportedType (exported, assembly.MainModule, di);
 					}
 				}
+
+				MarkFullyPreservedAssemblies ();
 			}
 
 			ProcessPendingTypeChecks ();
 		}
 
+		static bool IsFullyPreservedAction (AssemblyAction action) => action == AssemblyAction.Copy || action == AssemblyAction.Save;
+
+		void MarkFullyPreservedAssemblies ()
```

Comment:
I find the logic here odd. Why is this method called in the marking loop? What does it need to precompute all the references?

Reply:
See https://github.com/mono/linker/blob/feature/lazyload/docs/design/assembly-loading.md#exceptions-to-lazy-loading. It's called in the marking loop because marking can load new assemblies which may introduce references that would get the copy/save action.

Reply:
I filed https://github.com/mono/linker/issues/1800 to track the ideas we discussed which might help us get rid of this.